packages/snaps-jest/src/internals/request.ts

```diff
@@ -103,29 +114,94 @@ export function handleRequest({
 }
 
 /**
- * Get the response content either from the SnapInterfaceController or the response object if there is one.
+ * Get the interface ID from the result if it's available or create a new interface if the result contains static components.
  *
  * @param result - The handler result object.
  * @param snapId - The Snap ID.
  * @param controllerMessenger - The controller messenger.
- * @returns The content components if any.
+ * @returns The interface ID or undefined if the result doesn't include content.
  */
-export function getContentFromResult(
+export async function getInterfaceFromResult(
   result: unknown,
   snapId: SnapId,
   controllerMessenger: RootControllerMessenger,
-): Component | undefined {
+) {
   if (isPlainObject(result) && hasProperty(result, 'id')) {
-    return controllerMessenger.call(
-      'SnapInterfaceController:getInterface',
-      snapId,
-      result.id as string,
-    ).content;
+    return result.id as string;
   }
 
   if (isPlainObject(result) && hasProperty(result, 'content')) {
-    return result.content as Component;
+    const id = await controllerMessenger.call(
+      'SnapInterfaceController:createInterface',
+      snapId,
+      result.content as Component,
+    );
+
+    return id;
   }
 
   return undefined;
 }
+
+/**
+ * Get the response content from the SnapInterfaceController and include the interaction methods.
+ *
+ * @param result - The handler result object.
+ * @param snapId - The Snap ID.
+ * @param controllerMessenger - The controller messenger.
+ * @returns The content components if any.
+ */
+export async function getInterfaceApi(
+  result: unknown,
+  snapId: SnapId,
+  controllerMessenger: RootControllerMessenger,
+): Promise<(() => SnapHandlerInterface) | undefined> {
+  const interfaceId = await getInterfaceFromResult(
+    result,
+    snapId,
+    controllerMessenger,
+  );
+
+  if (interfaceId) {
+    return () => {
+      const { content } = controllerMessenger.call(
+        'SnapInterfaceController:getInterface',
+        snapId,
+        interfaceId,
+      );
+
+      const clickElementFn: SnapInterfaceActions['clickElement'] = async (
```

Comment:
It's a bit redundant to name this `...Fn`.
```suggestion
      const clickElement: SnapInterfaceActions['clickElement'] = async (
```

Reply:
`clickElement` is already defined in this file (it's a function)